src/services/log.ts

```diff
@@ -84,6 +85,34 @@ export class LogFormatter {
     return out
   }
 
+  private safeHTML(entry: any): any {
+    if (typeof entry == "string") {
+      return escape(entry)
+    }
+
+    entry = {...entry}
+    for (const key in entry) {
+      if (typeof entry[key] == "string") {
+        entry[key] = escape(entry[key])
+      }
+    }
```

Comment:
I think this needs to take nesting into account, of arrays and objects

Reply:
The recursive call `renderExpandedRecursively` iterates over the nested structure and calls this each time, but you have a good point, it doesn't handle arrays otherwise. Will fix in the next PR.